File: Data Structures & Algorithms/roman-to-integer/submission-3.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mp = {
            "M": 1000,
            "D": 500,
            "C": 100,
            "L": 50,
            "X": 10,
            "V": 5,
            "I": 1
        }
        res = 0
        for i, c in enumerate(s):
            if i < len(s) - 1 and mp[s[i]] < mp[s[i + 1]]:
                res -= mp[c]
            else:
                res += mp[c]
        return res

File: Data Structures & Algorithms/palindromic-substrings/submission-3.rs
impl Solution {
    pub fn count_substrings(s: String) -> i32 {
        let bytes = s.as_bytes();
        let mut res = 0i32;
        for center in 0..bytes.len() {
            res += Self::count_pali(bytes, 0, center, center);
            res += Self::count_pali(bytes, 0, center, center + 1);
        }
        res
    }
    fn count_pali(
        bytes: &[u8],
        mut count: i32,
        mut left: usize,
        mut right: usize
    ) -> i32 {
        while right < bytes.len() && bytes[left] == bytes[right] {
            count += 1;
            let Some(next_left) = left.checked_sub(1) else {
                return count;
            };
            left = next_left;
            right += 1
        }
        count
    }
}
